### PR TITLE
:wrench: Fix up a few more effc++ items

### PR DESCRIPTION
**Pull request by mattismyname**
test suite has these two failures, but the same tests fail even before this change:

21/70 Test #62: test-regression_all .................***Failed   12.56 sec
29/70 Test #61: test-regression_default .............***Failed   12.91 sec

### CODE REVIEW

**Comment by nlohmann** [comment]
The tests should succeed... Can you make sure you on the latest `develop` version?

**Comment by theodelrieu** [comment]
What are the errors?

**Comment by mattismyname** [comment]
Ok, my bad. The failures were with gcc 4.9.3. When run under 5.4.0-6ubuntu1~16.04.5 everything passes.

**Comment by nlohmann** [comment]
Hm... GCC 4.9 should work as well. Could you post the errors?

**Comment by nlohmann** [comment]
And which compiler created the `-Weffc++` warnings?

**Comment by mattismyname** [comment]
Original effc++ warnings were from icc (ICC) 11.0 20081105. I'll dig them up and post them here.

Sorry, never used ctest -- how can I get it to print the specific errors? I just get very general message currently:

```
$ ctest -R test-regression_default
Test project /nfs/orto/home/mkgumbel/clt_dev02/json/build
    Start 61: test-regression_default
1/1 Test #61: test-regression_default ..........***Failed    6.87 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
default    =   6.87 sec (1 test)

Total Test time (real) =   6.94 sec

The following tests FAILED:
         61 - test-regression_default (Failed)
Errors while running CTest

```
thanks

**Comment by nlohmann** [comment]
You can use `--output-on-failure`.

File: src/json.hpp [hunk 1]
```diff
@@ -6101,7 +6101,9 @@ class serializer
     @param[in] ichar  indentation character to use
     */
     serializer(output_adapter_t<char> s, const char ichar)
-        : o(std::move(s)), loc(std::localeconv()),
+        : o(std::move(s)),
+          number_buffer(),
```
Comment (nlohmann) [inline src/json.hpp]: What is the warning that led to adding `number_buffer()` here? The member is initialized in the definition:

```cpp
std::array<char, 64> number_buffer{{}};
```

File: src/json.hpp [hunk 2]
```diff
@@ -8011,6 +8013,7 @@ class basic_json
         json_value(number_float_t v) noexcept : number_float(v) {}
         /// constructor for empty values of a given type
         json_value(value_t t)
+            : object(nullptr)
```
Comment (nlohmann) [inline src/json.hpp]: If this is required, then maybe we can remove the lines

```cpp
object = nullptr;  // silence warning, see #821
```

below which have been added when discussing #821.

Comment (gregmarr) [reply src/json.hpp]: Related: line 8073 is now unreachable due to the case added at line 8062.

Comment (nlohmann) [reply src/json.hpp]: That case was always unreachable and was only added to get the hash `961c151d2e87f2686a955a9be24d316f1362bf21` into binaries using the library.

Comment (gregmarr) [reply src/json.hpp]: Interesting.  Perhaps a comment is in order the.

**Comment by nlohmann** [comment]
@mattismyname Could you comment on https://github.com/nlohmann/json/pull/858#pullrequestreview-81649476 and https://github.com/nlohmann/json/pull/858#pullrequestreview-81650164?

**Comment by nlohmann** [comment]
@mattismyname Could you please comment on https://github.com/nlohmann/json/pull/858#pullrequestreview-81649476 and https://github.com/nlohmann/json/pull/858#pullrequestreview-81650164? I would like to have this PR merged for the upcoming release.

File: src/json.hpp (hunk 1 above)
Comment (mattismyname) [reply src/json.hpp]: It was bad icc check -- I think this change is not really necessary.

File: src/json.hpp (hunk 2 above)
Comment (mattismyname) [reply src/json.hpp]: I think this can be removed -- it was bad icc check.

**Comment by mattismyname** [comment]
Sorry for slowness

File: src/json.hpp (hunk 1 above)
Comment (nlohmann) [reply src/json.hpp]: Thanks for checking back. Could you revert this change please?

File: src/json.hpp (hunk 2 above)
Comment (nlohmann) [reply src/json.hpp]: Thanks. Could you remove the mentioned line above please?

**Comment by mattismyname** [comment]
I've pushed the requested fixes to mattismyname/json develop branch.

**Comment by nlohmann** [comment]
Thanks a lot!

**Pull request merged**